boot/boot_serial/src/boot_serial.c

```diff
@@ -396,7 +410,27 @@ bs_upload(char *buf, int len)
     static struct flash_sector status_sector;
 #endif
 
-    img_num = 0;
+    zcbor_state_t zsd[4];
+    zcbor_new_state(zsd, sizeof(zsd) / sizeof(zcbor_state_t), (uint8_t *)buf, len, 1);
+
+    struct zcbor_map_decode_key_val image_upload_decode[] = {
+        ZCBOR_MAP_DECODE_KEY_DECODER("image", zcbor_uint32_decode, &img_num),
+        ZCBOR_MAP_DECODE_KEY_DECODER("data", zcbor_bstr_decode, &img_chunk_data),
+        ZCBOR_MAP_DECODE_KEY_DECODER("len", zcbor_size_decode, &img_size_tmp),
+        ZCBOR_MAP_DECODE_KEY_DECODER("off", zcbor_size_decode, &img_chunk_off),
+        ZCBOR_MAP_DECODE_KEY_DECODER("sha", zcbor_bstr_decode, &sha),
+        ZCBOR_MAP_DECODE_KEY_DECODER("upgrade", zcbor_bool_decode, &upgrade)
```

Comment:
We should probably #ifdef this line as it will not be that useful and popular in serial recovery.

Reply:
> If it's not here, won't the validation fail if a client sends it with their request?

Validation by zcbor_map_decode_bulk? The thing only searches for defined keys, so if you pass it a key it does not know, it ignores it. This has been intentionally done to work with extending protocol from client side, while server side lags behind.

Reply:
Ah was under the impression it did validation, will remove.

Reply:
Removed sha and upgrade